src/scopeHandler.js

```diff
@@ -84,12 +95,14 @@ export function createScopeHandler(unsafeRec, safeGlobal, sloppyGlobals) {
       // todo: allow modifications when target.hasOwnProperty(prop) and it
       // is writable, assuming we've already rejected overlap (see
       // createSafeEvaluatorFactory.factory). This TypeError gets replaced with
-      // target[prop] = value
-      if (objectHasOwnProperty(target, prop)) {
+      // endowments[prop] = value
+      if (prop in endowments) {
         // todo: shim integrity: TypeError, String
         throw new TypeError(`do not modify endowments like ${String(prop)}`);
       }
 
+      // Set the value on the shadow. The target itself is an empty
```

Comment:
`Shadow` term is a reference to the "Allegory of the Cave" from Plato,https://en.wikipedia.org/wiki/Allegory_of_the_Cave, specially this phrase:

> "The shadows are the prisoners' reality."

Reply:
There are two different **shadows** here, and although the comment should be fixed to clear the confusion, its statement was indeed correct.

1. The proxy target, when a proxy is used as a membrane, becomes a shadow target. Alex provides a good defintion:

> Shadow target: Every Proxy instance in ECMAScript requires a shadow target to form the base for the proxy. In es-membrane, this means the shadow target usually functions as a copy of the original object, with cached properties defined only as necessary. Shadow targets must be of the same primordial type as the object we're reflecting: to call a proxy as if it were a function, the proxy's shadow target must be a function. To make a proxy look like an array, the shadow target must be an array, and so on.

https://github.com/ajvincent/es-membrane

The allegory of the cave makes sense, since we refer to ***a process (the light) that projects reality on the wall of the cave***.

2. In the original realm-shim, the proxy target is an object made of the `endowments` that inherit from the `safeGlobal`. This places `safeGlobal` behind, or ***in the shadow*** of the endowments.

The dictionary definition of "shadow" makes sense since it refers to ***behind a dominating and pervasive presence***.

https://www.thefreedictionary.com/shadow